frontend/src/routes/questions/question-table.tsx

```diff
@@ -0,0 +1,200 @@
+    getPaginationRowModel: getPaginationRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getCoreRowModel: getCoreRowModel(),
+    onColumnFiltersChange: setColumnFilters,
+    onPaginationChange: setPagination,
+  });
+
+  const handleDifficultyFilterChange = (value: string) => {
+    setColumnFilters(value === 'all' ? [] : [{ id: 'difficulty', value }]);
+  };
+
+  const handleStatusFilterChange = (value: string) => {
+    setColumnFilters(value == 'all' ? [] : [{ id: 'attempted', value: value === 'attempted' }]);
+  };
+
+  return (
+    <div>
+      <div className='flex items-center py-4'>
+        <div className='mr-2'>
+          <Select onValueChange={handleStatusFilterChange}>
+            <SelectTrigger className='w-[140px]'>
+              <SelectValue placeholder='Status' />
+            </SelectTrigger>
+            <SelectContent>
+              <SelectItem value='all'>All</SelectItem>
+              <SelectItem value='attempted'>Attempted</SelectItem>
+              <SelectItem value='not-attempted'>Not attempted</SelectItem>
+            </SelectContent>
+          </Select>
+        </div>
+        <div className='mr-2'>
+          <Select onValueChange={handleDifficultyFilterChange}>
+            <SelectTrigger className='w-[140px]'>
+              <SelectValue placeholder='Difficulty' />
+            </SelectTrigger>
+            <SelectContent>
+              <SelectItem value='all'>All</SelectItem>
+              <SelectItem value='Easy'>Easy</SelectItem>
+              <SelectItem value='Medium'>Medium</SelectItem>
+              <SelectItem value='Hard'>Hard</SelectItem>
+            </SelectContent>
+          </Select>
+        </div>
+        <Input
+          placeholder='Search questions...'
+          value={(table.getColumn('title')?.getFilterValue() as string) ?? ''}
+          onChange={(event) => table.getColumn('title')?.setFilterValue(event.target.value)}
+          className='max-w-sm'
+        />
+      </div>
+      <div className='rounded-md border'>
+        <Table>
+          <TableHeader>
+            {table.getHeaderGroups().map((headerGroup) => (
+              <TableRow key={headerGroup.id}>
+                {headerGroup.headers.map((header) => {
+                  return (
+                    <TableHead key={header.id}>
+                      {header.isPlaceholder
+                        ? null
+                        : flexRender(header.column.columnDef.header, header.getContext())}
+                    </TableHead>
+                  );
+                })}
+              </TableRow>
+            ))}
+          </TableHeader>
+          <TableBody>
+            {!isError && table.getRowModel().rows?.length ? (
+              table.getRowModel().rows.map((row) => (
+                <TableRow key={row.id}>
+                  {row.getVisibleCells().map((cell) => (
+                    <TableCell key={cell.id}>
+                      <Link to={`/questions/${row.id}`}>
```

Comment:
Add hover underline styling and 1-based indexing instead (currently is off-by-one error)